### PR TITLE
RFE: Unit Display more enhancements implement

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
#3949 RFE: Unit Display more enhancements 

- add ability to set the panel order for the nontabbed version in the Client Settings
- save the size and position of the tabbed and nontabbed version independently

### CODE REVIEW

**Comment by kuronekochomusuke** [comment]
This is ready for review

File: megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java [hunk 1]
```diff
@@ -888,6 +936,42 @@ protected void okAction() {
             clientgui.updateButtonPanel(GamePhase.TARGETING);
         }
 
+        // unit display non tabbed
+        if (!GUIPreferences.getInstance().getDisplayStartTabbed()) {
+            boolean unitDisplayNonTabbedChanged = false;
+            int s = unitDisplayNonTabbed.getSize();
+
+            if ((s > 0) && (!unitDisplayNonTabbed.get(0).equals(UnitDisplayOrderPreferences.getInstance().getString("NonTabbedA1")))) {
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/CommonSettingsDialog.java:944]: I'm seeing a lot of repeated magic numbers (s > 0, .get(0)) and magic strings (e.g. "NonTabbedA1") here, can we put them into enums or constants instead? It'll make this easier to maintain going forward.

**Review by NickAragua — CHANGES_REQUESTED** [review]
Looks fine, just needs a few long term maintainability tweaks.

**Review by NickAragua — APPROVED** [review]
Good to go.

**Pull request merged**